crates/sovereign-sdk/rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -50,10 +53,11 @@ pub trait ZkvmHost: Zkvm + Clone {
     /// with some mild performance overhead and is not as easy to debug as [`simulate_with_hints`](ZkvmHost::simulate_with_hints).
     fn run(&mut self, with_proof: bool) -> Result<Proof, anyhow::Error>;
 
-    /// Extracts public input form the proof.
+    /// Extracts public input and receipt from the proof.
+    #[allow(clippy::type_complexity)]
     fn extract_output<Da: DaSpec, Root: BorshDeserialize>(
         proof: &Proof,
-    ) -> Result<StateTransition<Da, Root>, Self::Error>;
+    ) -> Result<(StateTransition<Da, Root>, Option<Receipt>), Self::Error>;
```

Comment:
I don't really like binding *Risc0* receipts here tbh. what if we use another zkvm?

we should keep these kind of trait generic

Reply:
Then we should be either accepting serializing the proof twice to not expose the Receipt, or should be exposing It by creating a `VerifiableReceipt` trait, and implementing the trait on the risc0::Receipt type.